### ISSUE TITLE
Feedbacks on the prototype

### ISSUE THREAD

**Issue by bluejay9676** [riceluxs1t/KBOPrediction#2]
===Model===
- Use Gradient, such as decent, newton, quasi-newton instead of Adam, as it isn't hard to get a derivative of a polynomial cost function http://www.benfrederickson.com/numerical-optimization/
- Normalize based on the team's score.
e.g)
TEAM_A scored 12 and TEAM_B scored 8. The hypothesis (the winning prob) will be 12/(12 + 8)
or 
TEAM_A scored 3 and TEAM_B scored 7. The hypothesis (the winning prob) will be 3/10
- Make a design variable and print it out before the cost function. You should be able to check the input and the hypothesis. 

===Data===
http://www.statiz.co.kr/main.php
1. 타율 방어율은 fidelity가 낮음.
2. Sabermetrics가 꽤 중요함
3. Importance: FIP/FIP+ >> ERA
4. OPS/OPS+ >> AVG

**Comment by riceluxs1t** [comment]
Questions
1. How is going from the tuple of two scores to a single probability justified? Is it the home team's winning probability? How do you predict the total score X + Y ? I.e. A score of 1 : 0 translates to the home team's likelihood of winning equal to 100%. This is counter intuitive.

2. ERA/HRA -> low fidelity. What is his logic behind this? Can he back this up with some numbers? 

3. How are those sabermetric terms calculated? Can they be computed based on the raw information we already have?

**Comment by bluejay9676** [comment]
1. True. I see some logical holes here and there. But I still think the current model's cost function allows outliers to affect the model significantly. This requires a lot of thinking.

2. The feedback mentioned that ERA/HRA are purely based on luck (not sure why, but I don't think it will hurt to try out a model that does not take in ERA/HRA as an input)

3. I assume so, but seems like http://www.statiz.co.kr/main.php already has them?

**Comment by riceluxs1t** [comment]
1. How does that conversion process help with the model's cost function dealing with outliers better? Outliers, however they are defined, can be analyzed separately and filtered out from the training data set. I don't see how your answer addresses any of my questions. Why not build your model to directly output a score? why normalize scores when the weight of each score is effectively the same across all games? I don't see your reasoning. Is this something that has to do with NN internals?

2. This sounds like typical a feature engineering part of data  science. Your friend probably has far more domain knowledge than us but actually verifying his claim or at least finding some references showing his result might be good. 

3. Sites like that typically provide data as of now. We need snapshots of those data at the time of each game. It would be good to have him included in on this issue page if he has some engineering background.

All in all, I don't see any data/math-supported reason to blindly trust such new metrics. If they are easy to compute given our current data, sure it is worth a try but if that's not the case, we should do some more research into it.

**Comment by bluejay9676** [comment]
In summary,

1. There will be minor changes in the structure of NN & cost function.

2. More research on meaningful data.

3. Maybe scraping more from statiz (this is the lowest priority for now).

**Comment by bluejay9676** [comment]
1. GradientwithClipping() from tensorflow. This will take care of the outliers
